doc/march_packages/march_hardware_interface.rst

```diff
@@ -1,10 +1,69 @@
-:orphan:
-
-.. todo:: make sure to remove this orphan tag when merging.
-
-.. todo:: (Isha) add documentation
-
 .. _march-hardware-interface-label:
 
 march_hardware_interface
 ========================
+The march_hardware_interface package uses the march_hardware (see
+:ref:`march-hardware-label`) library to create an interface between ros_control
+and the exoskeleton hardware. It has one runnable node which functions as the
+control loop. The loop consists of the following steps:
+
+#. Reading data from the robot
+#. Updating the connected controllers using the read data
+#. Writing the output of the controllers to the robot
+
+The class that is responsible for all this is the ``MarchHardwareInterface``.
+It implements the ``hardware_interface::RobotHW`` class from `ros_control`_.
+It implements three important functions from that base class: ``init``, ``read``
+and ``write``. Furthermore, the class owns an instance of the ``MarchRobot``
+class, which it uses for reading from and writing to the robot. The reading
+and writing must happen in real time to ensure it runs deterministically and,
+therefore, run at high frequencies. To guarantee this, real time publishers are
+used from `realtime_tools`_ and messages from ROS are published and received in
+a ``ros::AsyncSpinner``. Furthermore, all dynamically allocated resources, such
+as ``std::vector``, reserve their memory in the ``init`` function to guarantee
+no allocations take place during real time.
+
+.. _ros_control: https://wiki.ros.org/ros_control
+.. _realtime_tools: https://wiki.ros.org/realtime_tools
+
+Custom Controller Interfaces
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The march_hardware_interface package also includes two custom controller
```

Comment:
I agree. Perhaps call them just hardware interfaces? ros_control calls them controllers, so that's why.

Reply:
I decided to keep them at _controller interfaces_. Since that is how they are called in ros_control and hardware interface would be conflicting with the march hardware interface.